packages/ui/src/components/Table/Body/Body.tsx

```diff
@@ -1,12 +1,68 @@
-import React from 'react'
+import React, { Children } from 'react'
 import { Box } from '../../Box/Box'
+import { IconSolidLoading } from '../../icons/'
+import { Row } from '../Row/Row'
+import { Cell } from '../Cell/Cell'
+import { Text } from '../../Text/Text'
 
 import * as styles from './styles.css'
 
 export type Props = {
 	children: React.ReactNode
 }
 
-export const Body: React.FC<Props> = ({ children }) => {
-	return <Box cssClass={styles.body}>{children}</Box>
+// implicit props set from the parent table
+type ParentPassedProps = {
+	loading?: boolean
+	error?: string
 }
+
+export const Body: React.FC<Props & ParentPassedProps> = ({
+	children,
+	loading,
+	error,
+}) => {
+	const renderBodyContent = () => {
+		if (loading) {
+			return <LoadingBody />
+		}
+
+		if (error) {
+			return <ErrorBody error={error} />
+		}
+
+		const empty = Children.toArray(children).length === 0
+		if (empty) {
+			return <EmptyBody />
+		}
+
+		return children
+	}
+
+	return <Box cssClass={styles.body}>{renderBodyContent()}</Box>
+}
+
+const LoadingBody: React.FC = () => (
+	<Row gridColumns={['100%']}>
+		<Cell justifyContent="center" my="8">
+			<IconSolidLoading className={styles.loadingIcon} />
+			<Text>Loading...</Text>
+		</Cell>
+	</Row>
+)
+
+const ErrorBody: React.FC<{ error: string }> = ({ error }) => (
+	<Row gridColumns={['100%']}>
+		<Cell justifyContent="center" my="8">
+			<Text color="bad">{error}</Text>
+		</Cell>
+	</Row>
+)
+
+const EmptyBody: React.FC = () => (
+	<Row gridColumns={['100%']}>
+		<Cell justifyContent="center" my="8">
+			<Text>No results</Text>
+		</Cell>
+	</Row>
+)
```

Comment:
Nice to have these different states componentized, but wondering if we want to keep them hidden inside the implementation of `Table.Body`. I could see us wanting to customize these in the future. e.g. if we have an empty state, maybe we want to show a CTA to finish integrating.

What do you think about exporting these but forcing people to render them explicitly inside `Table.Body` so they can override them if needed? Also, maybe we provide something like `Table.NoContent` that provides the `Row` + `Cell` and takes in children for content to display in the center.

Nothing blocking here, just a few thoughts - take or leave them!